java/common/src/main/java/org/ray/util/MethodId.java

```diff
@@ -0,0 +1,199 @@
+package org.ray.util;
+
+import com.google.common.base.Preconditions;
+import java.io.Serializable;
+import java.lang.invoke.MethodHandleInfo;
+import java.lang.invoke.SerializedLambda;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.WeakHashMap;
+import org.objectweb.asm.Type;
+import org.ray.util.logger.RayLog;
+
+
+public final class MethodId {
+
+  /* use ThreadLocal to avoid lock. </br>
+   * A cache from the lambda instances to SerializedLambda.
+   * Note: the lambda instances are dynamically created per call site,
+   * we use WeakHashMap to avoid OOM.
+   */
+  private static final ThreadLocal<WeakHashMap<Class<Serializable>, MethodId>>
```

Comment:
same here, `RayFunc`. 
and a lot of other places in the rest of this PR.

Reply:
MethodId is not only for RayFunc, it clould used for all lambda call. the check of RayFunc has add in entrance

Reply:
MethodId is not only for RayFunc, it clould used for all lambda call. the check of RayFunc has add in entrance